src/ImpactEvaluator.sol

```diff
@@ -123,14 +123,20 @@ contract ImpactEvaluator is AccessControl, Balances {
         address payable[] memory addresses,
         uint64[] memory scores
     ) private {
+        uint addedBalance = 0;
+        uint balanceConsumed = 0;
+        uint scoreUnit = previousRoundRoundReward / MAX_SCORE;
         for (uint i = 0; i < addresses.length; i++) {
             address payable participant = addresses[i];
-            uint amount = (scores[i] * previousRoundRoundReward) / MAX_SCORE;
+            uint amount = scores[i] * scoreUnit;
```

Comment:
The old and the new versions may produce different values because they apply rounding at different stages.

A hypothetical example to illustrate the problem:

- `MAX_SCORE` is 100
- `previousRoundRoundReward` is `250`
- `score[i]` is `2`

Before:
```
amount = (2 * 250) / 100 = 500 / 100 = 5
```

After:
```
amount = 2 * (250/100) = 2 * 2 = 4
```

The situation is even worse when `previousRoundRoundReward` is less than `MAX_SCORE`:
- `MAX_SCORE` is 100
- `previousRoundRoundReward` is `50`
- `score[i]` is `2`

Before:
```
amount = (2 * 50) / 100 = 100 / 100 = 1
```

After:
```
amount = 2 * (50/100) = 2 * 0 = 0
```

Having written that, I don't remember the scale of `MAX_SCORE` and `previousRoundRoundRewards` numbers. Maybe we can prove that we will never run into these two scenarios.

Reply:
You're right, the remaining round reward can be quite low and this will greatly increase the error. I'll revert this one part